dot/state/epoch.go

```diff
@@ -191,21 +189,25 @@ func (s *EpochState) GetEpochForBlock(header *types.Header) (uint64, error) {
 		return 0, errors.New("header is nil")
 	}
 
-	firstSlot, err := s.baseState.loadFirstSlot()
+	if header.Number == 1 {
+		return 0, nil
+	}
+
+	veryFirstSlotNumber, err := s.retrieveFirstNonOriginBlockSlot(header.Hash())
 	if err != nil {
-		return 0, err
+		return 0, fmt.Errorf("retrieving very first slot number: %w", err)
 	}
 
 	slotNumber, err := header.SlotNumber()
 	if err != nil {
 		return 0, fmt.Errorf("getting slot number: %w", err)
 	}
 
-	return (slotNumber - firstSlot) / s.epochLength, nil
+	return (slotNumber - veryFirstSlotNumber) / s.epochLength, nil
```

Comment:
nit/perhaps a better name here. Maybe generisBlockSlotNumber or something like that? I think having a first slot and very first slot is a bit confusing

Reply:
the genesis block does not have a slot number, I can change to `chainFirstSlotNumber`